src/Agent.Worker/TestResults/Parser.cs

```diff
@@ -72,7 +72,9 @@ public class JUnitParser : Parser, IParser
         protected override ITestResultParser GetTestResultParser(IExecutionContext executionContext)
         {
             var traceListener = new CommandTraceListener(executionContext);
-            return new JUnitResultParser(traceListener);
+            var featureFlagService = executionContext.GetHostContext().GetService<IFeatureFlagService>();
+            var enableJunitAttachments = featureFlagService.GetFeatureFlagState(TestResultsConstants.JUnitTestCaseAttachmentsEnabled, TestResultsConstants.TCMServiceInstanceGuid);
+            return new JUnitResultParser(traceListener, false, enableJunitAttachments);
```

Comment:
Consider usage measurement for the new feature (perhaps in another PR)?

Reply:
Usage measurement is already in place via customer intelligence that tracks number of file attachments